src/components/user/UserProfile.jsx

```diff
@@ -0,0 +1,419 @@
+    });
+    const { updateProfile } = this.props;
+    e.preventDefault();
+    const {
+      email, username, bio, website, city, country, phone,
+      image, firstName, lastName,
+    } = this.state;
+
+    const userInfo = {
+      first_name: firstName,
+      last_name: lastName,
+      email,
+      image,
+      username,
+      bio,
+      website,
+      city,
+      country,
+      phone,
+    };
+    updateProfile(userInfo);
+  }
+
+  onAlertClick() {
+    this.setState({
+      showAlert: false,
+    });
+  }
+
+  render() {
+    const {
+      variant, showAlert, alertHeading, alertBody,
+    } = this.state;
+
+    const errorChecker = (field, errorMessage) => {
+      const { formErrorFields } = this.state;
+      if (formErrorFields.includes(field)) {
+        return errorMessage;
+      }
+      return '';
+    };
+
+    const {
+      formErrorFields, imageUrl, username, bio, firstName,
+      lastName, city, country, phone, email, website,
+    } = this.state;
+    return (
+      <div className="content">
+        <br />
+        <AlertComponent
+          showAlert={showAlert}
+          onAlertClick={this.onAlertClick}
+          alertBody={alertBody}
+          alertHeading={alertHeading}
+          variant={variant}
+        />
+        <Row>
+          <Col md="6">
+            <Card className="profile-card-user">
+              <div className="profile-card-user-items">
+                <CardBody>
+                  <CardText />
+                  <div className="author">
+                    <div className="card-photo">
+                      <img
+                        alt="..."
+                        className="avatar"
+                        src={imageUrl}
+                      />
+                    </div>
+                    <h5 className="title">{username}</h5>
+
+                    <p className="description">Author</p>
+                  </div>
+                  <div className="card-description">{bio}</div>
+                </CardBody>
+                <CardFooter className="card-footer">
+                  <hr />
+                  <UserStats
+                    articlesNum={20}
+                    followersNum={3}
+                    followingNum={2}
+                  />
+                  <hr />
+                  <p>
+                    <strong className="container">Articles Written</strong>
+                  </p>
+                  <Articles />
+                </CardFooter>
+              </div>
+            </Card>
+          </Col>
+          <Col md="6">
+            <Card className="profile-form">
+              <CardHeader>
+                <h5 className="title">Edit Your Profile</h5>
+              </CardHeader>
+              <CardBody>
+                <Form className="input-form" id="user-input-form">
+                  <Row>
+                    <Col className="pr-md-1" md="5">
+                      <FormGroup>
+                        <label id="username">
```

Comment:
also, fix this, JSX labels always need some other staff too.

Reply:
Yeah, this is what I was talking about on the call, no need to have control, ids, etc since we already have them inside input, this was raised as a bug withing esLint and they are working to resolve it, read more here `https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/302`